modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/FriendlyURLDLFileEntryLocalServiceWrapper.java

```diff
@@ -0,0 +1,188 @@
+					dlFileEntry.getFileEntryId());
+
+			for (FriendlyURLEntry friendlyURLEntry : friendlyURLEntries) {
+				_friendlyURLEntryLocalService.deleteFriendlyURLEntry(
+					friendlyURLEntry);
+			}
+		}
+
+		return dlFileEntry;
+	}
+
+	@Override
+	public DLFileEntry updateFileEntry(
+			long userId, long fileEntryId, String sourceFileName,
+			String mimeType, String title, String description, String changeLog,
+			DLVersionNumberIncrease dlVersionNumberIncrease,
+			long fileEntryTypeId, Map<String, DDMFormValues> ddmFormValuesMap,
+			File file, InputStream inputStream, long size, Date expirationDate,
+			Date reviewDate, ServiceContext serviceContext)
+		throws PortalException {
+
+		DLFileEntry dlFileEntry = super.updateFileEntry(
+			userId, fileEntryId, sourceFileName, mimeType, title, description,
+			changeLog, dlVersionNumberIncrease, fileEntryTypeId,
+			ddmFormValuesMap, file, inputStream, size, expirationDate,
+			reviewDate, serviceContext);
+
+		if (_ffFriendlyURLEntryFileEntryConfiguration.enabled()) {
+			_updateFriendlyURL(dlFileEntry, title);
+		}
+
+		return dlFileEntry;
+	}
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		_ffFriendlyURLEntryFileEntryConfiguration =
+			ConfigurableUtil.createConfigurable(
+				FFFriendlyURLEntryFileEntryConfiguration.class, properties);
+	}
+
+	private void _addFriendlyURLEntry(DLFileEntry dlFileEntry, String urlTitle)
+		throws PortalException {
+
+		String uniqueUrlTitle = _friendlyURLEntryLocalService.getUniqueUrlTitle(
+			dlFileEntry.getGroupId(),
+			_classNameLocalService.getClassNameId(FileEntry.class),
+			dlFileEntry.getFileEntryId(), urlTitle,
+			LanguageUtil.getLanguageId(LocaleUtil.getSiteDefault()));
+
+		_friendlyURLEntryLocalService.addFriendlyURLEntry(
+			dlFileEntry.getGroupId(),
+			_classNameLocalService.getClassNameId(FileEntry.class),
+			dlFileEntry.getFileEntryId(), uniqueUrlTitle,
+			ServiceContextThreadLocal.getServiceContext());
+	}
+
+	private void _updateFriendlyURL(DLFileEntry dlFileEntry, String urlTitle)
+		throws PortalException {
+
+		try {
+			FriendlyURLEntry friendlyURLEntry =
+				_friendlyURLEntryLocalService.getMainFriendlyURLEntry(
+					_classNameLocalService.getClassNameId(FileEntry.class),
+					dlFileEntry.getFileEntryId());
+
+			if (!Objects.equals(friendlyURLEntry.getUrlTitle(), urlTitle)) {
+				String uniqueUrlTitle =
+					_friendlyURLEntryLocalService.getUniqueUrlTitle(
+						dlFileEntry.getGroupId(),
+						_classNameLocalService.getClassNameId(FileEntry.class),
+						dlFileEntry.getFileEntryId(), urlTitle,
+						LanguageUtil.getLanguageId(
+							LocaleUtil.getSiteDefault()));
+
+				_friendlyURLEntryLocalService.
+					updateFriendlyURLEntryLocalization(
+						friendlyURLEntry,
+						LanguageUtil.getLanguageId(LocaleUtil.getSiteDefault()),
+						uniqueUrlTitle);
+			}
+		}
+		catch (Exception exception) {
+			_addFriendlyURLEntry(dlFileEntry, urlTitle);
+		}
```

Comment:
Unless you're forced to, it is better not to capture raw Exception types, but concrete ones (e.g. NoSuchEntryExn). In this case it would be even better to check if the friendly URL exists with one of the `fetch*` methods, and call `add*` or `update*` depending on the result. When possible, the use of exceptions should be exceptional (there are exceptions like DLApp, where there are no fetch methods).

Reply:
On this case the problem is that I did not found no `fetch*` method to look that for a className/classPk a friendlyURL exists if not with a particular urlTitle

I will catch NoSuchModelException because  `findByC_C` throws `NoSuchFriendlyURLEntryMappingException` and `findByPrimaryKey`  throws `NoSuchFriendlyURLEntryException` that are the methods called under `getMainFriendlyURLEntry`